internal/server/add_firm.go

```diff
@@ -26,16 +25,12 @@ type addFirmVars struct {
 }
 
 func renderTemplateForAddFirm(client FirmInformation, defaultPATeam int, tmpl Template) Handler {
-	return func(perm sirius.PermissionSet, w http.ResponseWriter, r *http.Request) error {
+	return func(perm sirius.PermissionSet, dd sirius.DeputyDetails, w http.ResponseWriter, r *http.Request) error {
 
 		ctx := getContext(r)
 		routeVars := mux.Vars(r)
 		deputyId, _ := strconv.Atoi(routeVars["id"])
-		deputyDetails, err := client.GetDeputyDetails(ctx, defaultPATeam, deputyId)
-
-		if err != nil {
-			return err
-		}
+		deputyDetails := dd
```

Comment:
This doesn't need to be reassigned. If you want `dd` to be called `deputyDetails`, name it as such in the function params.

The same applies for every file.